internal/dataplane/translator/atc/field.go

```diff
@@ -92,3 +93,30 @@ func (f HTTPQueryField) FieldType() FieldType {
 func (f HTTPQueryField) String() string {
 	return "http.queries." + f.QueryParamName
 }
+
+// HTTPPathSingleSegmentField represensts a single segment of HTTP path with 0 based index.
+type HTTPPathSingleSegmentField struct {
+	Index int
+}
+
+func (f HTTPPathSingleSegmentField) FieldType() FieldType {
+	return FieldTypeString
+}
+
+func (f HTTPPathSingleSegmentField) String() string {
+	return fmt.Sprintf("http.path.segments.%d", f.Index)
```

Comment:
Having some components as embedded values in function (sometimes repeated across multiple functions) and some as consts feels a bit off, but I guess the embedded values are easier to read.

Reply:
The defined constants are used for fields that has no parameters (like `http.method`,`net.protocol`).